xarray/core/dataarray.py

```diff
@@ -2312,11 +2316,14 @@ def quantile(self, q, dim=None, interpolation='linear', keep_attrs=False):
         numpy.nanpercentile, pandas.Series.quantile, Dataset.quantile
         """
 
+        if keep_attrs is None:
+            keep_attrs = _get_keep_attrs(default=False)
+
```

Comment:
The next two should not be necessary since `Dataset.quantile()` and `Dataset.rank()` have the same statement.